src/components/MapView/AlertForm/index.tsx

```diff
@@ -0,0 +1,354 @@
+  createStyles,
+  FormControl,
+  ListSubheader,
+  MenuItem,
+  Select,
+  TextField,
+  Theme,
+  Typography,
+  withStyles,
+  WithStyles,
+} from '@material-ui/core';
+import { grey } from '@material-ui/core/colors';
+
+import { faBell, faCaretDown } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+import { useDispatch, useSelector } from 'react-redux';
+
+import {
+  getBoundaryLayerSingleton,
+  LayerDefinitions,
+} from '../../../config/utils';
+import {
+  BoundaryLayerProps,
+  WMSLayerProps,
+  LayerKey,
+} from '../../../config/types';
+import { LayerData } from '../../../context/layers/layer-data';
+import { layerDataSelector } from '../../../context/mapStateSlice/selectors';
+import { availableDatesSelector } from '../../../context/serverStateSlice';
+import LayerDropdown from '../Layers/LayerDropdown';
+
+const boundaryLayer = getBoundaryLayerSingleton();
+
+function AlertForm({ classes }: AlertFormProps) {
+  const dispatch = useDispatch();
+  const boundaryLayerData = useSelector(layerDataSelector(boundaryLayer.id)) as
+    | LayerData<BoundaryLayerProps>
+    | undefined;
+
+  const availableDates = useSelector(availableDatesSelector);
+
+  const [isAlertFormFormOpen, setIsAlertFormFormOpen] = useState(false);
+
+  // form elements
+  const [hazardLayerId, setHazardLayerId] = useState<LayerKey>();
+  const [selectedDate, setSelectedDate] = useState<number | null>(null);
+  const [regionsList, setRegionsList] = useState<string[]>(['allRegions']);
+  const [emailValid, setEmailValid] = useState<boolean>(false);
+  const [email, setEmail] = useState<string>('');
+  const [selectedStat, setSelectedStat] = useState<string>('');
+  const [selectedThreshold, setSelectedThreshold] = useState<number>(0);
+
+  // Very basic regex for detecting emails
+  const emailRegex: RegExp = RegExp('.+@.+');
+
+  // On by default, may want to disable it in some cases
+  const isAlertFormEnabled = true;
+
+  // set default date after dates finish loading and when hazard layer changes
+  useEffect(() => {
+    const dates = hazardLayerId
+      ? availableDates[
+          (LayerDefinitions[hazardLayerId] as WMSLayerProps)?.serverLayerName
+        ]
+      : null;
+    if (!dates || dates.length === 0) {
+      setSelectedDate(null);
+    } else {
+      setSelectedDate(dates[dates.length - 1]);
+    }
+  }, [availableDates, hazardLayerId]);
+
+  const regionNamesToNsoCodes: Map<string, string> = useMemo(() => {
+    if (!boundaryLayerData) {
+      // Not loaded yet. Will proceed when it is.
+      return new Map();
+    }
+
+    return new Map(
+      boundaryLayerData.data.features
+        .filter(feature => feature.properties != null)
+        .map(feature => [
+          `${feature.properties?.ADM1_EN} / ${feature.properties?.ADM2_EN}`,
+          feature.properties?.NSO_CODE,
+        ]),
+    );
+  }, [boundaryLayerData]);
+
+  const regionNames: string[] = useMemo(() => {
```

Comment:
If you're using `useMemo` for performance, we likely don't need it here since the code is quite simple. Let me know if I'm wrong. 

If you wish to continue using them,  we should try combine this memo and the one above to output `{regionNames, regionNamesToNsoCodes}` since they share the same dependency array, and produce similar outputs (both produce `${feature.properties?.ADM1_EN} / ${feature.properties?.ADM2_EN}`)

Reply:
I will look at combining them; and it's true that the code is quite simple although the underlying dataset is quite large so it is pretty slow (at least anecdotally, on my machine) so I was hoping to only do it once. I'll take another look.

Reply:
I did not end up combining them, but I made them (hopefully) easier to understand.

The reason I have a separate list is because one of these is a sorted list of strings, and one is an object whose keys are those strings. Unfortunately I don't think there's really a way to have the object keys sorted in memory and the semantics of `.sort()` (since it's self-mutating) make me think doing this each render is probably wasteful.

If I am understanding `useMemo` and using it correctly, performance-wise it seems pretty good to me how it is. But I'm still happy to change this if you want.

Reply:
If it helps performance, all good (useMemo's indented use!) :+1: